### ISSUE TITLE
Working with Resources declaration

### ISSUE THREAD

**Issue by jbremmer** [99xt/serverless-dynamodb-local#56]
Wouldn't it be better if this plugin would work with Resources section from Serverless v.1 ?

That way I wouldn't have to define them twice in yml and in json for this plugin.

**Comment by AshanFernando** [comment]
This is a good suggestion. The challenge we need to solve is we need to create the local dynamodb tables also from the defined schema. @jbremmer Any suggestions?

**Comment by jbremmer** [comment]
Not sure what you mean, I would not want to have two definitions. YAML one is what I would want to work with possibly extending a custom section with seed file pointer.

**Comment by msjonker** [comment]
I've created a branch that uses the serverless.yml Resources section for creation of DynamoDB tables here: https://github.com/msjonker/serverless-dynamodb-local/tree/serverless-config

Unfortunately, since dynamodb-migrations is no longer used, seeding is not available. @AshanFernando if you could provide some feedback, I'd appreciate it.

**Comment by AshanFernando** [comment]
@msjonker I think we can replace dynamodb-migrations with a new library called dynamodb-seeds, only with seeds functionality. What do you think?

**Comment by msjonker** [comment]
Thanks for the feedback!

I can't find dynamodb-seeds; did you mean dynamo-seeder? If so, that library depends on a schema file existing for each table that is being seeded, which gets us back to where we were with a separate schema file having to be updated in tandem with the Serverless Resources definition.

To me, it feels that seeding should be separate from this plugin, as seeding could be used both for local or production instance of DynamoDB. It looks like this was attempted at one point in https://github.com/marc-hughes/serverless-dynamodb-fixtures-plugin, but has not been updated to work with Serverless v1+. What is described there is more like I was thinking. I don't use the seeding functionality of serverless-dynamodb-local, but would be willing to see what it takes to get that plugin up-to-date, if you are open to that solution.

**Comment by AshanFernando** [comment]
Hi @msjonker, even now serverless-dynamodb-local supports seeding in local as well as production environments. However the feature was missing in the documentation after creating a new branch for the serverless 1.0 version of the plugin. This we could fix easily. @mjzone Will you be able to look into this?

I agree with you. Seeding should be separated from the migration plugin. Since the migration feature is moving to yml, migration plugin is no longer needed. Now migrations more becoming a part of serverless (Previously we wrote functionality as third party plugins if its not depending on serverless structure). 

So my proposal is to create a new plugin called dynamodb-seeds extracting the seeding functionality from dynamodb-migrations(Both local and production seeding) and use with serverless-dynamodb-local to make the plugin up-to date with both yml based migrations and seeding functionality.

So shall we move on that directions so that we can update the plugins?

**Comment by msjonker** [comment]
Yep, sounds good!

**Comment by daviskoh** [comment]
I currently use serverless variables like so

serverless.yaml
```
resources:
  Resources:
    myTable:
      Type: AWS::DynamoDB::Table
      Properties:
        ${file(./dynamodb/my-table.json):Table}
```

**Comment by nbransby** [comment]
Where did you guys get to with this work? It seems like the the right approach to take

**Comment by msjonker** [comment]
I believe that the dynamodb-seeds plugin needs to be created. I haven't had a chance to work on that yet.

**Comment by christophgysin** [comment]
I reviewed, and tested this, works great and looks good!

I rebased it on top of v1, and pushed to my repo. Let me know if I can further help to get this merged.

**Comment by christophgysin** [comment]
This has been merged, and should be closed.

**Comment by msjonker** [comment]
Resources declaration addressed in #72. Seeding addressed in #77.
